README
NAME
    Plack::App::Env - Plack Env dump application.

SYNOPSIS
     use Plack::App::Env;

     my $obj = Plack::App::Env->new(%parameters);
     my $psgi_ar = $obj->call($env);
     my $app = $obj->to_app;

METHODS
  "new"
     my $obj = Plack::App::Env->new(%parameters);

    Constructor.

    Returns instance of object.

  "call"
     my $psgi_ar = $obj->call($env);

    Implementation of env dump.

    Returns reference to array (PSGI structure).

  "to_app"
     my $app = $obj->to_app;

    Creates Plack application.

    Returns Plack::Component object.

EXAMPLE
     use strict;
     use warnings;

     use Plack::App::Env;
     use Plack::Runner;

     # Run application.
     my $app = Plack::App::Env->new->to_app;
     Plack::Runner->new->run($app);

     # Output:
     # HTTP::Server::PSGI: Accepting connections at http://0:5000/

     # > curl http://localhost:5000/
     # \ {
     #     HTTP_ACCEPT            "*/*",
     #     HTTP_HOST              "localhost:5000",
     #     HTTP_USER_AGENT        "curl/7.64.0",
     #     PATH_INFO              "/",
     #     psgi.errors            *main::STDERR  (read/write, layers: unix perlio),
     #     psgi.input             *HTTP::Server::PSGI::$input  (layers: scalar),
     #     psgi.multiprocess      "",
     #     psgi.multithread       "",
     #     psgi.nonblocking       "",
     #     psgi.run_once          "",
     #     psgi.streaming         1,
     #     psgi.url_scheme        "http",
     #     psgi.version           [
     #         [0] 1,
     #         [1] 1
     #     ],
     #     psgix.harakiri         1,
     #     psgix.input.buffered   1,
     #     psgix.io               *Symbol::GEN1  (read/write, layers: unix perlio),
     #     QUERY_STRING           "",
     #     REMOTE_ADDR            "127.0.0.1",
     #     REMOTE_PORT            39562,
     #     REQUEST_METHOD         "GET",
     #     REQUEST_URI            "/",
     #     SCRIPT_NAME            "",
     #     SERVER_NAME            0,
     #     SERVER_PORT            5000,
     #     SERVER_PROTOCOL        "HTTP/1.1"
     # }

DEPENDENCIES
    Data::Printer, Plack::Component.

REPOSITORY
    <https://github.com/michal-josef-spacek/Plack-App-Env>

AUTHOR
    Michal Josef Špaček <mailto:skim@cpan.org>

    <http://skim.cz>

LICENSE AND COPYRIGHT
    © 2020-2022 Michal Josef Špaček

    BSD 2-Clause License

VERSION
    0.10



Changes
0.10

0.09 2022-09-10T18:18:30+02:00
 - Rename example to better name.
 - Update copyright years.

0.08 2022-02-27T18:03:29+01:00
 - Simplify failing test.

0.07 2020-12-09T18:43:36+01:00
 - Fix comment in doc example.

0.06 2020-11-03T14:27:09+01:00
 - Fix test for development version of Data::Printer.

0.05 2020-11-02T13:54:26+01:00
 - Fix test »REMOTE_ADDR "127.0.0.1" (dualvar: 127),« on FreeBSD.

0.04 2020-10-30T14:19:02+01:00
 - Fix test (psgi.errors has different values).

0.03 2020-10-29T16:05:46+01:00
 - Add test.
 - Change bugtracker to github.

0.02 2020-08-07T20:03:40+02:00
 - Add examples/ dir to META.yml.
 - Fix test dependencies.
 - Regen README file.
 - Update examples/ex1.pl example.

0.01 2020-08-07T19:59:05+02:00
 - First version.


Makefile.PL
use lib '.';
use strict;
use warnings;

use inc::Module::Install;

# Definition.
abstract 'Plack env dump application.';
author 'Michal Josef Spacek <skim@cpan.org>';
author_requires 'English' => 0;
author_requires 'File::Object' => 0;
author_requires 'Test::More' => 0;
author_requires 'Test::NoWarnings' => 0;
author_requires 'Test::Pod' => 0;
author_requires 'Test::Pod::Coverage' => 0;
license 'bsd';
name 'Plack-App-Env';
perl_version 5.008;
readme_from 'Env.pm';
recursive_author_tests('xt');
requires 'Data::Printer' => 0;
requires 'Plack::Component' => 0;
resources 'bugtracker' => 'https://github.com/michal-josef-spacek/Plack-App-Env/issues';
resources 'homepage' => 'https://github.com/michal-josef-spacek/Plack-App-Env';
resources 'repository' => 'git://github.com/michal-josef-spacek/Plack-App-Env';
test_requires 'Error::Pure' => 0;
test_requires 'File::Object' => 0;
test_requires 'Test::More' => 0;
test_requires 'Test::NoWarnings' => 0;
tests_recursive;
version '0.10';

# Run.
WriteAll();


Env.pm
package Plack::App::Env;

use base qw(Plack::Component);
use strict;
use warnings;

use Data::Printer;

our $VERSION = 0.10;

sub call {
	my ($self, $env) = @_;

	my $env_output;
	p $env, 'output' => \$env_output;

	return [
		200,
		['Content-Type' => 'text/plain'],
		[$env_output],
	];
}

1;

__END__

=pod

=encoding utf8

=head1 NAME

Plack::App::Env - Plack Env dump application.

=head1 SYNOPSIS

 use Plack::App::Env;

 my $obj = Plack::App::Env->new(%parameters);
 my $psgi_ar = $obj->call($env);
 my $app = $obj->to_app;

=head1 METHODS

=head2 C<new>

 my $obj = Plack::App::Env->new(%parameters);

Constructor.

Returns instance of object.

=head2 C<call>

 my $psgi_ar = $obj->call($env);

Implementation of env dump.

Returns reference to array (PSGI structure).

=head2 C<to_app>

 my $app = $obj->to_app;

Creates Plack application.

Returns Plack::Component object.

=head1 EXAMPLE

=for comment filename=plack_app_env.pl

 use strict;
 use warnings;

 use Plack::App::Env;
 use Plack::Runner;

 # Run application.
 my $app = Plack::App::Env->new->to_app;
 Plack::Runner->new->run($app);

 # Output:
 # HTTP::Server::PSGI: Accepting connections at http://0:5000/

 # > curl http://localhost:5000/
 # \ {
 #     HTTP_ACCEPT            "*/*",
 #     HTTP_HOST              "localhost:5000",
 #     HTTP_USER_AGENT        "curl/7.64.0",
 #     PATH_INFO              "/",
 #     psgi.errors            *main::STDERR  (read/write, layers: unix perlio),
 #     psgi.input             *HTTP::Server::PSGI::$input  (layers: scalar),
 #     psgi.multiprocess      "",
 #     psgi.multithread       "",
 #     psgi.nonblocking       "",
 #     psgi.run_once          "",
 #     psgi.streaming         1,
 #     psgi.url_scheme        "http",
 #     psgi.version           [
 #         [0] 1,
 #         [1] 1
 #     ],
 #     psgix.harakiri         1,
 #     psgix.input.buffered   1,
 #     psgix.io               *Symbol::GEN1  (read/write, layers: unix perlio),
 #     QUERY_STRING           "",
 #     REMOTE_ADDR            "127.0.0.1",
 #     REMOTE_PORT            39562,
 #     REQUEST_METHOD         "GET",
 #     REQUEST_URI            "/",
 #     SCRIPT_NAME            "",
 #     SERVER_NAME            0,
 #     SERVER_PORT            5000,
 #     SERVER_PROTOCOL        "HTTP/1.1"
 # }

=head1 DEPENDENCIES

L<Data::Printer>,
L<Plack::Component>.

=head1 REPOSITORY

L<https://github.com/michal-josef-spacek/Plack-App-Env>

=head1 AUTHOR

Michal Josef Špaček L<mailto:skim@cpan.org>

L<http://skim.cz>

=head1 LICENSE AND COPYRIGHT

© 2020-2022 Michal Josef Špaček

BSD 2-Clause License

=head1 VERSION

0.10

=cut
